src/Microsoft.DncEng.SecretManager/SecretTypes/RandomBase64.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DncEng.CommandLineLib;
+
+namespace Microsoft.DncEng.SecretManager.SecretTypes
+{
+    [Name("random-base64")]
+    public class RandomBase64 : SecretType<RandomBase64.Parameters>
+    {
+        private readonly ISystemClock _clock;
+
+        public class Parameters
+        {
+            public int Bytes { get; set; }
+        }
+
+        public RandomBase64(ISystemClock clock)
+        {
+            _clock = clock;
+        }
+
+        public override List<string> GetCompositeSecretSuffixes()
+        {
+            return new List<string>
+            {
+                "-primary",
+                "-secondary",
+            };
+        }
+
+        public override async Task<List<SecretData>> RotateValues(Parameters parameters, RotationContext context, CancellationToken cancellationToken)
+        {
+            string currentPrimary = await context.GetSecretValue(context.SecretName + "-primary");
```

Comment:
Just a minor thing, I would extract "-primary" as a constant as it's used twice.